internal/operands/node-labeller/resources.go

```diff
@@ -0,0 +1,319 @@
+	const clusterRoleName = "kubevirt-node-labeller"
+	return clusterRoleName + "-" + namespace
+}
+
+func ClusterRoleBindingName(namespace string) string {
+	return ServiceAccountName + "-" + namespace
+}
+
+func newClusterRole(namespace string) *rbac.ClusterRole {
+	return &rbac.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   ClusterRoleName(namespace),
+			Labels: commonLabels,
+		},
+		Rules: []rbac.PolicyRule{{
+			APIGroups: []string{""},
+			Resources: []string{"nodes"},
+			Verbs:     []string{"get", "update", "patch"},
+		}},
+	}
+}
+
+func newServiceAccount(namespace string) *core.ServiceAccount {
+	return &core.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceAccountName,
+			Namespace: namespace,
+		},
+	}
+}
+
+func newClusterRoleBinding(namespace string) *rbac.ClusterRoleBinding {
+	return &rbac.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ClusterRoleBindingName(namespace),
+		},
+		RoleRef: rbac.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     ClusterRoleName(namespace),
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+		Subjects: []rbac.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      ServiceAccountName,
+			Namespace: namespace,
+		}},
+	}
+}
+
+func newConfigMap(namespace string) *core.ConfigMap {
+	return &core.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ConfigMapName,
+			Namespace: namespace,
+		},
+		Data: configMapData,
+	}
+}
+
+func kubevirtNodeLabellerSleeperContainer() *core.Container {
+	// Build the kubevirtNodeLabellerSleeper Container
+
+	name := "kubevirt-node-labeller-sleeper"
+	image := common.GetNodeLabellerImage()
+	cmd := []string{"sleep"}
+	args := []string{"infinity"}
+	container := &core.Container{
+		Name:    name,
+		Image:   image.Sleeper,
+		Command: cmd,
+		Args:    args,
+	}
+	return container
+}
+
+func initContainerKvmInfoNfdPlugin() *core.Container {
+	// Build the KvmInfoNfdPlugin Init Container
+
+	name := "kvm-info-nfd-plugin"
+	image := common.GetNodeLabellerImage()
+	cmd := []string{"/bin/sh", "-c"}
+	args := []string{"cp /usr/bin/kvm-caps-info-nfd-plugin /etc/kubernetes/node-feature-discovery/source.d/;"}
+
+	container := &core.Container{
+		Name:            name,
+		Image:           image.KVMInfoNFD,
+		Command:         cmd,
+		Args:            args,
+		ImagePullPolicy: core.PullAlways,
+		VolumeMounts: []core.VolumeMount{
+			{
+				Name:      nfdVolumeName,
+				MountPath: nfdVolumeMountPath,
+			},
+		},
+	}
+	return container
+}
+
+func initContainerKubevirtCpuNfdPlugin() *core.Container {
+	// Build the KubevirtCpuNfdPlugin Init Container
+
+	name := "kubevirt-cpu-nfd-plugin"
+	image := common.GetNodeLabellerImage()
+	cmd := []string{"/bin/sh", "-c"}
+	args := []string{"cp /plugin/dest/cpu-nfd-plugin /etc/kubernetes/node-feature-discovery/source.d/;", "cp /config/cpu-plugin-configmap.yaml /etc/kubernetes/node-feature-discovery/source.d/cpu-plugin-configmap.yaml;"}
```

Comment:
All the variables can be inlined, but maybe it would be clearer if this long string is not inlined.

Reply:
Done